tarantool/types.py

```diff
@@ -0,0 +1,141 @@
+"""
+Additional Tarantool type definitions.
+"""
+
+import typing
+from dataclasses import dataclass
+
+@dataclass
+class BoxError():
```

Comment:
Why not `tarantool/msgpack_ext/types/error.py` ?

Reply:
Because it's actually not a msgpack extension type. You can get `BoxError` object as additional info (type is dictionary, not MP_EXT N) in plain error, and I think it would be 99% of cases when one will work with `BoxError`. But you also can get MP_EXT 3 type with the same dictionary data.